add language widget

Jibec opened the pull request:
when we have a multilangual website, we want to allow the user to switch language

Jibec committed:
allow usage of ltr

since v0.67.1 it is possible to define language direction
see https://github.com/gohugoio/hugo/issues/6550

Jibec committed:
allow usage of ltr

since v0.67.1 it is possible to define language direction
see https://github.com/gohugoio/hugo/issues/6550

Jibec committed:
default html direction: use auto instead of ltr

Co-authored-by: Munif Tanjim <muniftanjim@gmail.com>

Jibec committed:
replace rtl support by LanguageDirection

since v0.67.1 it is possible to define language direction
see gohugoio/hugo#6550

Commit:
add [i18n]: ar (Arabic) translation

Jibec committed:
add backward compatibility with rtl

Co-authored-by: Munif Tanjim <muniftanjim@gmail.com>

Jibec committed:
add language widget

when we have a multi-linguage website, we want to allow the user to switch language

Jibec committed:
add language widget

when we have a multi-linguage website, we want to allow the user to switch language

Jibec committed:
display content translation

if a traduction exists for the current page, display an icon, the language code and an URL

Jibec committed:
display content translation

if a traduction exists for the current page, display an icon, the language code and an URL

Jibec committed:
add mastodon in social menu

mastodon can take any url
mastodon uses a mecanism to prove you are owner of something: that's the reason of the rel=me usage

Jibec committed:
allow header/extra

Jibec committed:
Merge remote-tracking branch 'origin/mentions' into current

Jibec committed:
don't display language icons if no translations

Jibec committed:
add license parameter

it's better to have an explicit mention to license in each page so people are not confused

Jibec committed:
Remove colon from template

Jibec committed:
convert i18n files into json

Weblate does support json files: https://docs.weblate.org/en/latest/formats.html#go-i18n-json-files

rffontenelle committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (1 of 1 strings)

Translation: Languages in floss/website-theme
Translate-URL: https://translate.fedoraproject.org/projects/languages-in-floss/website-theme/pt_BR/

Jibec committed:
move license before date

makes it easier to read article translations

milotype committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (1 of 1 strings)

Translation: Languages in floss/website-theme
Translate-URL: https://translate.fedoraproject.org/projects/languages-in-floss/website-theme/hr/

Commit:
Translated using Weblate (Polish)

Currently translated at 100.0% (1 of 1 strings)

Translation: Languages in floss/website-theme
Translate-URL: https://translate.fedoraproject.org/projects/languages-in-floss/website-theme/pl/

weblate committed:
Added translation using Weblate (Portuguese (Portugal))

weblate committed:
Added translation using Weblate (Occitan)

troyane commented:
@Jibec @MunifTanjim Is this PR supposed to be merged?
Cause on a way to get such functionality on my own site I've came up with this solution: https://github.com/troyane/minimo/commit/669ebec4d3749982fa7bddf986e9ca4d78388dff 

Does it make sense to create separate PR in case if this will not be merged?

weblate committed:
Added translation using Weblate (Finnish)

Ricky-Tigg committed:
Translated using Weblate (Finnish)

Currently translated at 100.0% (1 of 1 strings)

Translation: Languages in floss/website-theme
Translate-URL: https://translate.fedoraproject.org/projects/languages-in-floss/website-theme/fi/

weblate committed:
Added translation using Weblate (Sinhala)

weblate committed:
Added translation using Weblate (Korean)

simmon-nplob committed:
Translated using Weblate (Korean)

Currently translated at 100.0% (1 of 1 strings)

Translation: Languages in floss/website-theme
Translate-URL: https://translate.fedoraproject.org/projects/languages-in-floss/website-theme/ko/

weblate committed:
Added translation using Weblate (Greek)

weblate committed:
Added translation using Weblate (Chinese (Traditional))

weblate committed:
Added translation using Weblate (Burmese)

weblate committed:
Added translation using Weblate (Czech)

matjazjeran committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (1 of 1 strings)

Translation: Languages in floss/website-theme
Translate-URL: https://translate.fedoraproject.org/projects/languages-in-floss/website-theme/sl/

xeanhort committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (1 of 1 strings)

Translation: Languages in floss/website-theme
Translate-URL: https://translate.fedoraproject.org/projects/languages-in-floss/website-theme/ca/

Jibec committed:
Hugo 0.73.0 compatibility

Jibec committed:
Hugo 0.120.0 compatibility

Jibec committed:
Hugo 0.140.3 compatibility

Jibec committed:
Merge pull request #2 from languages-in-floss/current

Current

Jibec committed:
Merge pull request #1 from weblate/weblate-languages-in-floss-website-theme

Translations update from Weblate

Jibec committed:
Merge pull request #3 from languages-in-floss/current

Current

Jibec committed:
add generation date in meta

Jibec committed:
add alt on logo

Jibec committed:
remove empty files

Commit:
Added translation using Weblate (Estonian)

Added translation using Weblate (Interlingua)

Added translation using Weblate (Russian)

Co-authored-by: Languages add-on <noreply-addon-languages@weblate.org>

igorbounov committed:
Translated using Weblate (Russian)

Currently translated at 100.0% (1 of 1 strings)

Co-authored-by: Igor Gorbounov <igor.gorbounov@gmail.com>
Translate-URL: https://translate.fedoraproject.org/projects/languages-in-floss/website-theme/ru/
Translation: Languages in floss/website-theme

Jibec committed:
Merge pull request #5 from weblate/weblate-languages-in-floss-website-theme

Translations update from Fedora Weblate

Jibec committed:
Merge pull request #6 from languages-in-floss/current

Current
